essentials-framework/Essentials Devices Common/Essentials Devices Common/VideoCodec/CiscoCodec/CiscoSparkCodec.cs

```diff
@@ -853,6 +862,14 @@ void DeserializeResponse(string response)
 					{
 						GetBookings(null);
 					}
+					
+					else if (response.IndexOf("\"UserInterface\":{") > -1 || response.IndexOf("\"UserInterface\": {") > -1) // External Source Trigger
+					{
+						CiscoCodecEvents.RootObject eventReceived = new CiscoCodecEvents.RootObject();
+                        JsonConvert.PopulateObject(response, eventReceived);
+						Debug.Console(2, this, "*** Got an External Source Selection {0} {1}", eventReceived, eventReceived.Event.UserInterface, eventReceived.Event.UserInterface.Presentation.ExternalSource.Selected.SourceIdentifier.Value);
+						RunRouteAction(eventReceived.Event.UserInterface.Presentation.ExternalSource.Selected.SourceIdentifier.Value, null);

```

Comment:
In the interest of defensive programming, it's probably best to check if `RunRouteAction` is null prior to trying to invoke it.

Reply:
Agreed, added.